src/backend/api/meals.js

```diff
@@ -2,13 +2,119 @@ const express = require("express");
+    let sortDir = req.query.sortDir;
+
+    const response = {
+      data: [],
+      status: 200,
+      message: "the message",
+    };
+
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+
+    // api/meals?maxPrice=90
+    if (maxPrice) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("price", "<", +maxPrice);
+
+      response.data = result;
+    }
+
+    //api/meals?availableReservations=true
+    if (availableReservations===true) {
+      const result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) < max_reservations'));
+
+      response.data = result;
+    } else {
+      result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) > max_reservations'));
+
+      response.data = result;
+    }
+
+    //api/meals?title=Rød%20grød
+    if (title) {
+      const result = await knex("meals")
+      .select("*")
+      .where("title", "like", `%${title}%`);
+
+      response.data = result;
+    }
+
+    //api/meals?dateAfter=2022-10-01
+    if (dateAfter) {
+      const result = await knex
+        .select("*")
+        .from("meals")
+        .where("meal_time", ">", new Date(dateAfter));
+
+      response.data = result;
+    }
+
+    //api/meals?dateBefore=2022-08-08
+    if (dateBefore) {
+      const result = await knex("meals")
+      .select("*")
+      .where("meal_time", "<", dateBefore);
+
+      response.data = result;
+    }
+
+    //api/meals?limit=7
+    if (limit) {
+      const result = await knex("meal")
+        .select("*")
+        .limit(+limit);
+
+      response.data = result;
+    }
+
+    //api/meals?sortKey=price
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      const result = await knex("meal")
+        .select("*")
+        .orderBy(sortKey, sortDir === "desc" ? sortDir : "asc");
+
+      response.data = result;
+    }
+
+    //api/meals?sortKey=price&sortDir=desc
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+
+      if (sortDir === "asc" || sortDir === "desc") {
+        const result = await knex("meal")
+          .select("*")
+          .orderBy(sortKey, sortDir);
+
+        response.data = result;
+      }
+    }
+
+    res.status(response.status).json({ data: response.data, message: response.message });
```

Comment:
At the end you just return the response data, whatever it ended up being throughout the function. That is great! However, that means that only one of the filters can be applied to the `GET` method at once. The exercise however asked that these filters can be combined, like `?maxPrice=60&title=Spaghetti`.

This might seem challenging at first, but actually `knex` allows to do this in quite a clean manner. 🧠 

You can just start off your method with the basic query that you WOULD execute, which is:
```javascript
let resultQuery = knex("meals");
```

NOTICE how we do **not** actually call `select()` on it, or `await` it. We are just building the query for now.

In subsequent `if` statements, if a given query parameter is supplied, for instance `maxPrice`, you can add the relevant query to the current query:

```javascript
if (maxPrice && !isNaN(maxPrice)) {
    resultQuery = resultQuery.where("price", "<", +maxPrice);
 }
```